application/src/test/java/org/togetherjava/tjbot/commands/tophelper/TopHelperMessageListenerTest.java

```diff
@@ -0,0 +1,194 @@
+        assertFalse(shouldBeIgnored);
+    }
+
+    @Test
+    @DisplayName("Ignores bots")
+    void ignoresBots() {
+        // GIVEN a message from a bot
+        MessageReceivedEvent event =
+                createFakeMessageRecievedEvent(true, false, true, OVERVIEW_CHANNEL_PATTERN);
+
+        // WHEN checking if the message should be ignored
+        boolean shouldBeIgnored = topHelpersListenerSpy.shouldIgnoreMessage(event);
+
+        // THEN the message is ignored
+        assertTrue(shouldBeIgnored);
+    }
+
+    @Test
+    @DisplayName("Ignores webhooks")
+    void ignoresWebhooks() {
+        // GIVEN a message from a webhook
+        MessageReceivedEvent event =
+                createFakeMessageRecievedEvent(false, true, true, OVERVIEW_CHANNEL_PATTERN);
+
+        // WHEN checking if the message should be ignored
+        boolean shouldBeIgnored = topHelpersListenerSpy.shouldIgnoreMessage(event);
+
+        // THEN the message is ignored
+        assertTrue(shouldBeIgnored);
+    }
+
+    @Test
+    @DisplayName("Ignores wrong channels")
+    void ignoresWrongChannels() {
+        // GIVEN a message outside a help thread
+        MessageReceivedEvent eventNotAThread =
+                createFakeMessageRecievedEvent(false, false, false, OVERVIEW_CHANNEL_PATTERN);
+        MessageReceivedEvent eventWrongParentName =
+                createFakeMessageRecievedEvent(false, false, true, "memes");
+
+        // WHEN checking if the message should be ignored
+        boolean ignoresNonThreadChannels =
+                topHelpersListenerSpy.shouldIgnoreMessage(eventNotAThread);
+        boolean ignoresWrongParentNames =
+                topHelpersListenerSpy.shouldIgnoreMessage(eventWrongParentName);
+
+        // THEN the message is ignored
+        assertTrue(ignoresNonThreadChannels, "Failed to ignore non-thread channels");
+        assertTrue(ignoresWrongParentNames, "Failed to ignore wrong parent channel names");
+    }
+
+
+    MessageReceivedEvent createFakeMessageRecievedEvent(boolean isBot, boolean isWebhook,
+            boolean isThread, String parentChannelName) {
+        Message messageMock = mock(Message.class);
+        ThreadChannelImpl threadMock = mock(ThreadChannelImpl.class, RETURNS_DEEP_STUBS);
+
+        User user = new UserImpl(123456789, (JDAImpl) jdaTester.getJdaMock()).setName("John Doe")
+            .setDiscriminator("1234")
+            .setBot(isBot);
+
+        when(threadMock.getType())
+            .thenReturn(isThread ? ChannelType.GUILD_PUBLIC_THREAD : ChannelType.TEXT);
+        when(threadMock.getParentChannel().getName()).thenReturn(parentChannelName);
+        when(threadMock.asThreadChannel()).thenReturn(threadMock);
+
+        when(messageMock.isWebhookMessage()).thenReturn(isWebhook);
+        when(messageMock.getAuthor()).thenReturn(user);
+        when(messageMock.getChannel()).thenReturn(threadMock);
+
+        return new MessageReceivedEvent(jdaTester.getJdaMock(), 0, messageMock);
```

Comment:
i dont like that. the `jdaTester` should be expanded instead. that said, many of the things u do here are already covered by the jda tester

Reply:
Could you please explain how this should be done exactly?

Reply:
ur doing extensive jda mocking here. the point of `jdaTester` is to provide that for u. if u take a look at it, it for example has a method to create mocked users. `jdaTester.mockUser(...)` or sth like that.

so if u have a use case, such as creating a thread, u should add this logic to that class instead.

ur test shouldnt have to deal with details of jda implementations.